concreate/tools.py

```diff
@@ -32,18 +32,23 @@ def cleanup(target):
             shutil.rmtree(d)
 
 
-def load_descriptor(descriptor_path):
+def load_descriptor(descriptor):
     """ parses descriptor and validate it against requested schema type
 
     Args:
-      descriptor_path - path to image/modules descriptor to be validated
+      descriptor - yaml descriptor or path to a descriptor to be loaded
 
     Returns descriptor as a dictionary
     """
-    logger.debug("Loading descriptor from path '%s'." % descriptor_path)
+    if not os.path.exists(descriptor):
+        logger.info("Descriptor path '%s' doesn't exists, trying to parse it directly."
+                    % descriptor)
+        try:
+            return yaml.safe_load(descriptor)
+        except Exception:
+            raise ConcreateError('Cannot load descriptor.')
```

Comment:
Probably try to make it more understandable of what happened and why it could fail. We need to be clear about this.